### PR TITLE
Create Coordinator CLI

### LINKED ISSUE #48: Demo for Signature generation: Coordinator CLI
## Motivation

We want to simulate a simple execution of FROST to show potential FROST implementers how to use the FROST library.

## Scope

Build a simple CLI to demo signature aggregation by a coordinator in a 2-round FROST signature generation protocol, given a number of key shares and commitments from signature participants. 

All other actions that the coordinator entity would perform, prior to signature aggregation, during the FROST signature generation can be done by the person running the FROST demo, e.g by entering in the relevant details when prompted by the participant CLI instances in ZcashFoundation/frost#239.

### Round 2: Signature Aggregation

To perform signature aggregation, the CLI will prompt for the following data to be entered:
- The list of signature shares, `sig_shares`, for each participant in the signature generation.
- The message, `msg`, to be signed
- The list of commitments, `commitment_list`.

The CLI will first validate each signature share, `sig_share_i`, by running `DeserializeScalar`. 

If all signature shares are valid, the Coordinator CLI can proceed to aggregate them and produce the final signature by running, `aggregate`, with the previously entered data (`sig_shares`, `msg` and `commitment_list`) to generate the final Schnorr signature, `(R,z)` and output this to the terminal.

Two-Round FROST Signing Protocol with Coordinator - Round 2

### PR DESCRIPTION
**Pull request by natalieesk**
This closes #48

### CODE REVIEW

File: coordinator/src/step_3.rs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    participants: ParticipantsConfig,
+    signing_package: SigningPackage,
+) -> Result<Signature, Box<dyn std::error::Error>> {
+    writeln!(
+        logger,
+        "The number of signers: {}",
+        participants.participants.len()
+    )?;
+    println!("{}", participants.participants.len());
+
```
Comment (conradoplg) [inline coordinator/src/step_3.rs:38]: Is this needed?

```suggestion
```

File: coordinator/src/tests/values.md
Comment (conradoplg) [inline coordinator/src/tests/values.md:1]: These values are not working, I did a successful run using the values from the test, log as follows. So you probably just need to copy those values over

```
=== STEP 1: CHOOSE PARTICIPANTS ===

Paste the JSON public key package: 
{"signer_pubkeys":{"0300000000000000000000000000000000000000000000000000000000000000":"2cff4148a2f965801fb1f25f1d2a4e5df2f75b3a57cd06f30471c2c774419a41","0100000000000000000000000000000000000000000000000000000000000000":"fc2c9b8e335c132d9ebe0403c9317aac480bbbf8cbdb1bc3730bb68eb60dadf9"},"group_public":"15d21ccd7ee42959562fc8aa63224c8851fb3ec85a3faf66040d380fb9738673","ciphersuite":"FROST(Ed25519, SHA-512)"}
The number of participants: 
2
Identifier for participant 1 (hex encoded):
"0100000000000000000000000000000000000000000000000000000000000000"
Identifier for participant 2 (hex encoded):
"0300000000000000000000000000000000000000000000000000000000000000"
Selected participants:
"0100000000000000000000000000000000000000000000000000000000000000"
"0300000000000000000000000000000000000000000000000000000000000000"
=== STEP 2: CHOOSE MESSAGE AND GENERATE COMMITMENT PACKAGE ===

The message to be signed
74657374
The number of signers: 
Please enter JSON encoded commitments for participant Identifier(
    "0100000000000000000000000000000000000000000000000000000000000000",
):
{"hiding":"5078f5c6d679654bb88a8887242d49cc21a553ed26caed4d52570c6656fb9b92","binding":"936b660d3008d8298b0a7220a327a0813ffedd9d07604bdc73d7cffef63c0da0","ciphersuite":"FROST(Ed25519, SHA-512)"}
Please enter JSON encoded commitments for participant Identifier(
    "0300000000000000000000000000000000000000000000000000000000000000",
):
{"hiding":"91c2469b501fe5af8493f9ae77c8f57999460af317f2d9f2d4378ae0e665860e","binding":"c225618accff2266a45d87dc3219b04c774ca26c8629c4fa483e7e87da820007","ciphersuite":"FROST(Ed25519, SHA-512)"}
Signing Package: {"signing_commitments":{"0100000000000000000000000000000000000000000000000000000000000000":{"hiding":"5078f5c6d679654bb88a8887242d49cc21a553ed26caed4d52570c6656fb9b92","binding":"936b660d3008d8298b0a7220a327a0813ffedd9d07604bdc73d7cffef63c0da0","ciphersuite":"FROST(Ed25519, SHA-512)"},"0300000000000000000000000000000000000000000000000000000000000000":{"hiding":"91c2469b501fe5af8493f9ae77c8f57999460af317f2d9f2d4378ae0e665860e","binding":"c225618accff2266a45d87dc3219b04c774ca26c8629c4fa483e7e87da820007","ciphersuite":"FROST(Ed25519, SHA-512)"}},"message":"74657374","ciphersuite":"FROST(Ed25519, SHA-512)"}
=== STEP 3: BUILD GROUP SIGNATURE ===

The number of signers: 2
2
Please enter JSON encoded signatures for participant Identifier("0100000000000000000000000000000000000000000000000000000000000000"):
{"share":"b97409beff18861f0959530db091a64b812e3fefaa87e1e3d2c039f11d96cc09","ciphersuite":"FROST(Ed25519, SHA-512)"}
Please enter JSON encoded signatures for participant Identifier("0300000000000000000000000000000000000000000000000000000000000000"):
{"share":"9816a14e7cdecfcb240976f564cf98c5640e596b6ddf270379efbef4e9f7db0b","ciphersuite":"FROST(Ed25519, SHA-512)"}
Group signature: "72c948a63797c693e8e978fdb703a1f5a7590472a539da13b71dd6c2b8c1b2a664b7b4af6194439357c5d15f366760fce53c985a186709e74bb0f8e5078ea805"
=== END ===
```

**Review by conradoplg — CHANGES_REQUESTED** [review]
This is great, thanks! Just a couple of improvements to make

File: coordinator/src/tests/values.md
Comment (natalieesk) [reply coordinator/src/tests/values.md:1]: Ah I didn't update the message in the test values. Ok, thanks. I'll update

File: coordinator/src/step_3.rs (hunk 1 above)
Comment (natalieesk) [reply coordinator/src/step_3.rs:38]: Probably not. I'll remove

File: coordinator/src/tests/values.md
Comment (natalieesk) [reply coordinator/src/tests/values.md:1]: Done in [9f2c617](https://github.com/ZcashFoundation/frost-zcash-demo/pull/59/commits/9f2c617dab141cf89d37d92d0e0ef5f8d14ff537)

File: coordinator/src/step_3.rs (hunk 1 above)
Comment (natalieesk) [reply coordinator/src/step_3.rs]: Done in [9f2c617](https://github.com/ZcashFoundation/frost-zcash-demo/pull/59/commits/9f2c617dab141cf89d37d92d0e0ef5f8d14ff537)

File: coordinator/src/step_2.rs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+) -> Result<SigningPackage, Box<dyn std::error::Error>> {
+    writeln!(logger, "The message to be signed")?;
+
+    let mut msg = String::new();
+    input.read_line(&mut msg)?;
+
+    let message = hex::decode(msg.trim())?;
+
+    writeln!(logger, "The number of signers: ")?;
+
```
[the remark this answers is not in the document]
Comment (natalieesk) [reply coordinator/src/step_2.rs]: Done in [9f2c617](https://github.com/ZcashFoundation/frost-zcash-demo/pull/59/commits/9f2c617dab141cf89d37d92d0e0ef5f8d14ff537)

File: coordinator/src/step_1.rs [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+
+    for i in 1..=num_of_participants {
+        let package = pub_key_package.clone();
+        writeln!(logger, "Identifier for participant {:?} (hex encoded):", i).unwrap();
+
+        let mut identifier_input = String::new();
+
+        input.read_line(&mut identifier_input).unwrap();
+
+        let id_value = serde_json::from_str(&identifier_input).unwrap();
```
[the remark this answers is not in the document]
Comment (natalieesk) [reply coordinator/src/step_1.rs]: Done in [9f2c617](https://github.com/ZcashFoundation/frost-zcash-demo/pull/59/commits/9f2c617dab141cf89d37d92d0e0ef5f8d14ff537)

**Review by conradoplg — APPROVED** [review]

**Pull request merged**